pkg/runtime/core/trainingruntime_test.go

```diff
@@ -857,6 +857,54 @@ func TestTrainingRuntimeNewObjects(t *testing.T) {
 					Obj(),
 			},
 		},
+		"succeeded to build JobSet with imagePullSecrets overrides from the TrainJob's PodSpecOverrides.": {
```

Comment:
@toVersus @astefanutti @tenzen-y Do we want to keep separate tests for each PodSpecOverride field or combine them to reduce duplication?
Would it be easier to tests all fields under `succeeded to build JobSet with TrainJob's PodSpecOverrides` test.
https://github.com/kubeflow/trainer/blob/fab19aaf0cd445e4683a084e88144d5329769de8/pkg/runtime/core/trainingruntime_test.go#L182C4-L182C92

Reply:
@andreyvelich I agree that would make sense to combine them. If that's ok with you maybe we can merge this one and create an issue to refactor the PodSpecOverrides tests, WDYT?

Reply:
Sure, that sounds good.
@toVersus If you want, you can create a tracking issue for it.